bndtools.m2e/src/bndtools/m2e/MavenDependenciesRepository.java

```diff
@@ -24,37 +23,39 @@
 })
 public class MavenDependenciesRepository extends MavenWorkspaceRepository {
 
-	private final static Logger			logger	= LoggerFactory.getLogger(MavenDependenciesRepository.class);
+	private final static Logger					logger		= LoggerFactory
+		.getLogger(MavenDependenciesRepository.class);
+	private final static MavenBndrunContainer	mbc			= new MavenBndrunContainer();
 
 	private final Set<String>			allScopes	= new HashSet<>(
 		Arrays.asList("compile", "provided", "runtime", "system", "test"));
-	private final MavenBndrunContainer	mbc		= new MavenBndrunContainer();
 
 	@Override
 	public String getName() {
 		return "Maven Dependencies";
 	}
 
 	@Override
-	Supplier<Set<File>> collect(IMavenProjectFacade projectFacade, IProgressMonitor monitor) {
-		return () -> {
-			try {
-				Set<File> files = new HashSet<>();
-				MavenProject mavenProject = projectFacade.getMavenProject(monitor);
-				BndrunContainer bndrunContainer = mbc.getBndrunContainer(projectFacade, null, true, true, allScopes,
-					monitor);
-				DependencyResolver dependencyResolver = bndrunContainer.getDependencyResolver(mavenProject);
-				files.addAll(dependencyResolver.resolve()
-					.keySet());
-
-				logger.debug("{}: Collected files {} for project {}", getName(), files, projectFacade.getFullPath());
-
-				return files;
-			} catch (Exception e) {
-				logger.error("{}: Failed to collect files for project {}", getName(), projectFacade.getFullPath(), e);
-				return Collections.emptySet();
-			}
-		};
+	Optional<Supplier<Set<File>>> collect(IMavenProjectFacade projectFacade, IProgressMonitor monitor) {
+		logger.debug("{}: Collecting files for project {}", getName(), projectFacade.getProject());
```

Comment:
This changes the code from doing work in the future (when the supplier is called) to doing work now and returning a simple supplier. I assume you mean this change to do the work now? If so, then just return a `Set<File>`. The caller can check the set for emptyness and do other things including making the simple supplier. Or a the least change to return `Optional<Set<File>>`. Having a supplier in the return implies it is actually doing something interesting in the future. But with this change, all the interesting work has already been done.

Reply:
fixed